src/stash/src/postgres.rs

```diff
@@ -122,9 +123,22 @@ impl Postgres {
             // automatically seeded from OsRng", which meets this requirement.
             nonce: rand::random(),
         };
-        // Do the initial connection once here so we don't get stuck in transact's
-        // retry loop if the url is bad.
-        conn.connect().await?;
+        // Do the initial connection once here so we don't get stuck in
+        // transact's retry loop if the url is bad. Special case handling of
+        // retryable serialization errors.
+        loop {
+            let res = conn.connect().await;
+            if let Err(StashError {
+                inner: InternalStashError::Postgres(err),
+            }) = res
+            {
+                if let Some(&SqlState::T_R_SERIALIZATION_FAILURE) = err.code() {
```

Comment:
What is a retryable serialization error?

Reply:
A cockroach thing. Added a link to the doc about it and more explanation about what this is.